Contest_API.md

```diff
@@ -1032,9 +1032,9 @@ Properties of problem objects:
 | ordinal           | number    | A unique number that determines the order the problems, e.g. on the scoreboard.
 | rgb               | string ?  | Hexadecimal RGB value of problem color as specified in [HTML hexadecimal colors](https://en.wikipedia.org/wiki/Web_colors#Hex_triplet), e.g. `#AC00FF` or `#fff`.
 | color             | string ?  | Human readable color description associated to the RGB value.
-| time\_limit       | number ?  | Time limit in seconds per test data set (i.e. per single run). Should be an integer multiple of `0.001`.
-| test\_data\_count | integer ? | Number of test data sets.
-| max_score         | number ?  | Maximum expected score, although teams may score higher in some cases. Typically used to indicate scoreboard cell color in scoring contests. Required iff contest:scoreboard_type is `score`.
+| time\_limit       | number    | Time limit in seconds per test data set (i.e. per single run). Should be an integer multiple of `0.001`.
+| test\_data\_count | integer   | Number of test data sets.
```

Comment:
For both, if the server declares them in `/access`, then it really should be able to provide this information for *each* problem, so no point in allowing this to be nullable.

Reply:
Could there be individual problems where one or both of these don't make sense?

Reply:
I honestly think there are lots of cases where 'it wasn't provided to us' or 'we haven't configured that yet' (not necessarily just this property) and that's exactly _why_ we should keep properties like this nullable (or just missing/not set yet). We've had lots of data sets over the years that are incomplete or take time to fix, and there's a difference between 'it would be dumb to start a contest like this' and 'this state shouldn't be allowed to ever exist and your system must be built to enforce it'.

Reply:
_Maybe_ there are problems where a test data count or timelimit doesn't make sense, but I honestly cannot think of one. Also, I think `test_data_count = 0` is a perfectly reasonably response that provides more information than `null`.

If we don't have this information available intially or for certain problems, then I think that was/is a bug at that time and needs fixing, so the fact that it then is not according to spec, I see as a correct reflection of what's happening. In other words, if I'd then run a validation check against the API, I want to see this get flagged.